torch/csrc/jit/passes/quantization.cpp

```diff
@@ -2709,6 +2716,66 @@ void ReplicateDeQuant(std::shared_ptr<Graph>& graph) {
   }
 }
 
+void ReplicateChooseQParamsQuant(std::shared_ptr<Graph>& graph) {
+  std::stack<Block*> blocks_to_visit;
+  std::vector<Node*> quant_nodes_to_rewrite;
+  std::vector<Node*> choose_qparam_nodes_to_rewrite;
+  blocks_to_visit.push(graph->block());
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
```

Comment:
I think it's better to check for the call sequence of "choose_qparams - quant - dequant" here directly

Reply:
you can use subgraph matcher here as well

Reply:
At this point we have already called replicateDequant so the dequant nodes have been replicated. This way we can re-use that code from static and add this as a special case for dynamic